src/dist.rs

```diff
@@ -0,0 +1,308 @@
+use crate::{default_build_command, default_dist_dir, metadata};
+use anyhow::{ensure, Context, Result};
+use clap::Parser;
+use std::{fs, path::PathBuf, process};
+use wasm_bindgen_cli_support::Bindgen;
+
+/// An helper to generate the distributed package
+///
+/// This structs provide a customizable way to assemble and generate a package
+/// for Wasm.
+///
+/// # Usage
+///
+/// ```rust,no_run
+/// # use std::process;
+/// # use xtask_wasm::{anyhow::Result, clap};
+/// #
+/// # #[derive(clap::Parser)]
+/// # struct Opt {
+/// #     #[clap(subcommand)]
+/// #     cmd: Command,
+/// # }
+/// #
+/// #[derive(clap::Parser)]
+/// enum Command {
+///     Dist(xtask_wasm::Dist),
+/// }
+///
+/// fn main() -> Result<()> {
+///     let opt: Opt = clap::Parser::parse();
+///
+///     match opt.cmd {
+///         Command::Dist(dist) => {
+///             let dist = dist
+///                 .dist_dir_path("dist")
+///                 .static_dir_path("project/static")
+///                 .app_name("project")
+///                 .run_in_workspace(true)
+///                 .run("project")?;
+///
+///             println!("Built at {}", dist.dist_dir.display());
+///         }
+///     }
+///
+///     Ok(())
+/// }
+/// ```
+///
+/// Add a `dist` subcommand to assemble the distributed package for the
+/// `project` crate.
+///
+/// I will run the [`default_build_command`] at the workspace root, generate JS
+/// bindings and output two files: *project.js* and *project.wasm*.
+/// These two files will be write, and the content of the *project/static*
+/// directory will be copied, into the *dist* directory (returned by
+/// [`DistResult`]).
```

Comment:
Documentation doesn't use the pronoun "I" :joy: 

spelling:
"will be write" => "will be written"

Use backticks instead of `*` for paths and file names.

Don't use backticks in this case though:
"into the *dist* directory" => "into the dist directory"